server/service/integration_logger_test.go

```diff
@@ -0,0 +1,187 @@
+			assert.Equal(t, "/api/v1/fleet/config", kv["uri"])
+			assert.Equal(t, "admin1@example.com", kv["user"])
+		case 2:
+			assert.Equal(t, "info", kv["level"])
+			assert.Equal(t, "POST", kv["method"])
+			assert.Equal(t, "/api/v1/fleet/queries", kv["uri"])
+			assert.Equal(t, "admin1@example.com", kv["user"])
+			assert.Equal(t, "somequery", kv["name"])
+			assert.Equal(t, "select 1 from osquery;", kv["sql"])
+		default:
+			t.Fail()
+		}
+	}
+}
+
+func (s *integrationLoggerTestSuite) TestOsqueryEndpointsLogErrors() {
+	t := s.T()
+
+	_, err := s.ds.NewHost(&fleet.Host{
+		DetailUpdatedAt: time.Now(),
+		LabelUpdatedAt:  time.Now(),
+		SeenTime:        time.Now(),
+		NodeKey:         t.Name() + "1234",
+		UUID:            "1",
+		Hostname:        "foo.local",
+		OsqueryHostID:   t.Name(),
+		PrimaryIP:       "192.168.1.1",
+		PrimaryMac:      "30-65-EC-6F-C4-58",
+	})
+	require.NoError(t, err)
+
+	requestBody := io.NopCloser(bytes.NewBuffer([]byte(`{"node_key":"1234","log_type":"status","data":[}`)))
+	req, _ := http.NewRequest("POST", s.server.URL+"/api/v1/osquery/log", requestBody)
+	client := &http.Client{}
+	_, err = client.Do(req)
+	require.Nil(t, err)
+
+	logString := s.buf.String()
+	assert.Equal(t, `{"err":"decoding JSON: invalid character '}' looking for beginning of value","level":"info","path":"/api/v1/osquery/log"}
+`, logString)
+}
+
+func (s *integrationLoggerTestSuite) TestSubmitStatusLog() {
+	t := s.T()
+
+	_, err := s.ds.NewHost(&fleet.Host{
+		DetailUpdatedAt: time.Now(),
+		LabelUpdatedAt:  time.Now(),
+		SeenTime:        time.Now(),
+		NodeKey:         t.Name() + "1234",
+		UUID:            "1",
+		Hostname:        "foo.local",
+		PrimaryIP:       "192.168.1.1",
+		PrimaryMac:      "30-65-EC-6F-C4-58",
+		OsqueryHostID:   t.Name(),
+	})
+	require.NoError(t, err)
+
+	req := submitLogsRequest{
+		NodeKey: "1234",
+		LogType: "status",
+		Data:    nil,
+	}
+	res := submitLogsResponse{}
+	s.DoJSON("POST", "/api/v1/osquery/log", req, http.StatusOK, &res)
+
+	logString := s.buf.String()
+	assert.Equal(t, 1, strings.Count(logString, "\"ip_addr\""))
+	assert.Equal(t, 1, strings.Count(logString, "x_for_ip_addr"))
+}
+
+func (s *integrationLoggerTestSuite) TestEnrollAgentLogsErrors() {
+	t := s.T()
+	_, err := s.ds.NewHost(&fleet.Host{
+		DetailUpdatedAt: time.Now(),
+		LabelUpdatedAt:  time.Now(),
+		SeenTime:        time.Now(),
+		NodeKey:         "1234",
+		UUID:            "1",
+		Hostname:        "foo.local",
+		PrimaryIP:       "192.168.1.1",
+		PrimaryMac:      "30-65-EC-6F-C4-58",
+	})
+	require.NoError(t, err)
+
+	j, err := json.Marshal(&enrollAgentRequest{
+		EnrollSecret:   "1234",
+		HostIdentifier: "4321",
+		HostDetails:    nil,
+	})
+	require.NoError(t, err)
+
+	requestBody := io.NopCloser(bytes.NewBuffer(j))
+	req, _ := http.NewRequest("POST", s.server.URL+"/api/v1/osquery/enroll", requestBody)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	require.NoError(t, err)
+	require.NoError(t, resp.Body.Close())
```

Comment:
Maybe `testing_client.go` would need an option for `DoRaw` without authentication, as I believe this is the only difference with this block ^^ ?